### PR TITLE
fix: [#338] enable the action only if we have ProviderManager

### PR DESCRIPTION
**Pull request by BeckerFrank**
See #338 for Details:

I add enablement to the action and start the plugin during start so that the action is in the correct state.

### CODE REVIEW

**Comment by merks** [comment]
Does the bundle that sets the system property necessarily start?

**Review by ruspl-afed — CHANGES_REQUESTED** [review]
Thank you for working on this @BeckerFrank 

1) It is good practice to deliver format changes in a separate PR
2) I believe we can offer better way to implement the fix: involving system properties while working in a context of Eclipse Platform is excessive. Since property tester was already employed why not to implement one.
3) Starting the bundle manually may result in hours of debugging in the future, please avoid this.
and
4) The good fix would be to replace Bugzilla with GitHub, not just disable this action.

**Comment by BeckerFrank** [comment]
> Thank you for working on this @BeckerFrank
> 
> 1. It is good practice to deliver format changes in a separate PR

I only insert some new lines, the format change was not intended so I do not know how to avoid this.
> 2. I believe we can offer better way to implement the fix: involving system properties while working in a context of Eclipse Platform is excessive. Since property tester was already employed why not to implement one.

Ok, I try to find out how to do this.
> 3. Starting the bundle manually may result in hours of debugging in the future, please avoid this.
>    and
> 4. The good fix would be to replace Bugzilla with GitHub, not just disable this action.

This is an other task, because actual we have not removed the Bugzilla Stuff. I well take a closer look into the GitHub Connector and try to implement this.

**Comment by ruspl-afed** [comment]
> Ok, I try to find out how to do this.

You can have a look [here](https://help.eclipse.org/latest/index.jsp?topic=%2Forg.eclipse.platform.doc.isv%2Freference%2Fapi%2Forg%2Feclipse%2Fcore%2Fexpressions%2FPropertyTester.html) for example

File: mylyn.tasks/org.eclipse.mylyn.tasks.bugs/plugin.xml [hunk 1]
```diff
@@ -31,30 +31,54 @@
...
-                       name="org.eclipse.mylyn.tasks.bugs.actions.ReportBugAction.active"
-                       value="true">
-                 </systemProperty>
-              </enablement> 
-          </action>
-        </actionSet>
+   <extension
+         point="org.eclipse.ui.commands">
+      <command
+            description="description for Report Bug Action"
```
[the remark this answers is not in the document]
Comment (BeckerFrank) [reply mylyn.tasks/org.eclipse.mylyn.tasks.bugs/plugin.xml:37]: Done with next commit

File: mylyn.tasks/org.eclipse.mylyn.tasks.bugs/plugin.xml [hunk 2]
```diff
@@ -31,30 +31,54 @@
...
    </extension>
-  
+   <extension
+         point="org.eclipse.ui.handlers">
+       <handler
+            class="org.eclipse.mylyn.internal.tasks.bugs.actions.ReportBugHandler"
+            commandId="org.eclipse.mylyn.tasks.bugs.commands.ReportBugAction">
+        <enabledWhen>
+          <test
+                property="org.eclipse.mylyn.tasks.tasks.bugs.propertyTester.showTrim">
```
[the remark this answers is not in the document]
Comment (BeckerFrank) [reply mylyn.tasks/org.eclipse.mylyn.tasks.bugs/plugin.xml:49]: Removed with next commit

[the remark this answers is not in the document]
Comment (BeckerFrank) [reply mylyn.tasks/org.eclipse.mylyn.tasks.bugs/plugin.xml:49]: provider.any did not work so I use anyProvider with next commit

File: mylyn.tasks/org.eclipse.mylyn.tasks.bugs/plugin.xml [hunk 3]
```diff
@@ -31,30 +31,54 @@
...
+       <menuContribution
+            locationURI="menu:help?after=group.main.ext">
+            <command
+                  commandId="org.eclipse.mylyn.tasks.bugs.commands.ReportBugAction"
+                  icon="icons/etool16/task-bug-new.gif"
+                  mnemonic="%commands.newTaskFromMarker.mnemonic"
+                  tooltip="%ReportBugAction.tooltip"
+                  style="push">
+	      	<visibleWhen
+	        	    checkEnabled="true">
```
[the remark this answers is not in the document]
Comment (BeckerFrank) [reply mylyn.tasks/org.eclipse.mylyn.tasks.bugs/plugin.xml:76]: Yes with next commit

File: mylyn.tasks/org.eclipse.mylyn.tasks.bugs/src/org/eclipse/mylyn/internal/tasks/bugs/ProviderPropertyTester.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
+/*******************************************************************************
+ * Copyright © 2024 frank
```
[the remark this answers is not in the document]
Comment (BeckerFrank) [reply mylyn.tasks/org.eclipse.mylyn.tasks.bugs/src/org/eclipse/mylyn/internal/tasks/bugs/ProviderPropertyTester.java:2]: Sorry my fault changed with next commit.

File: mylyn.tasks/org.eclipse.mylyn.tasks.bugs/src/org/eclipse/mylyn/internal/tasks/bugs/ProviderPropertyTester.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+ *
+ * Contributors:
+ *     frank - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.mylyn.internal.tasks.bugs;
+
+import org.eclipse.core.expressions.PropertyTester;
+
+public class ProviderPropertyTester extends PropertyTester {
```
[the remark this answers is not in the document]
Comment (BeckerFrank) [reply mylyn.tasks/org.eclipse.mylyn.tasks.bugs/src/org/eclipse/mylyn/internal/tasks/bugs/ProviderPropertyTester.java:16]: Done with next commit.

File: mylyn.tasks/org.eclipse.mylyn.tasks.bugs/src/org/eclipse/mylyn/internal/tasks/bugs/ProviderPropertyTester.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+	@Override
+	public boolean test(Object receiver, String property, Object[] args, Object expectedValue) {
+		// ignore
```
[the remark this answers is not in the document]
Comment (BeckerFrank) [reply mylyn.tasks/org.eclipse.mylyn.tasks.bugs/src/org/eclipse/mylyn/internal/tasks/bugs/ProviderPropertyTester.java:20]: Good catch, changed with next commit

File: mylyn.tasks/org.eclipse.mylyn.tasks.bugs/src/org/eclipse/mylyn/internal/tasks/bugs/actions/ReportBugHandler.java [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+		Shell shell = HandlerUtil.getActiveShellChecked(event);
+		WizardDialog dialog = new WizardDialog(shell, new ReportBugOrEnhancementWizard());
+		dialog.open();
+		return null;
+	}
+
+	@Override
+	public void setEnabled(Object evaluationContext) {
+		super.setEnabled(evaluationContext);
+		SupportProviderManager providerManager = TasksBugsPlugin.getTaskErrorReporter().getProviderManager();
```
[the remark this answers is not in the document]
Comment (BeckerFrank) [reply mylyn.tasks/org.eclipse.mylyn.tasks.bugs/src/org/eclipse/mylyn/internal/tasks/bugs/actions/ReportBugHandler.java:38]: setEnabled is removed in the next commit.

File: mylyn.tasks/org.eclipse.mylyn.tasks.bugs/src/org/eclipse/mylyn/internal/tasks/bugs/TasksBugsPlugin.java [hunk 8]
```diff
@@ -78,6 +78,9 @@ public TasksBugsPlugin() {
 	public void start(BundleContext context) throws Exception {
 		super.start(context);
 		INSTANCE = this;
+		SupportProviderManager providerManager = getTaskErrorReporter().getProviderManager();
```
Comment (ruspl-afed) [inline mylyn.tasks/org.eclipse.mylyn.tasks.bugs/src/org/eclipse/mylyn/internal/tasks/bugs/TasksBugsPlugin.java:81]: With property tester employed we can remove this code.

File: mylyn.tasks/org.eclipse.mylyn.tasks.bugs/plugin.xml [hunk 9]
```diff
@@ -20,33 +20,63 @@
...
-              id="org.eclipse.mylyn.tasks.ui.bug.report"
-              label="%ReportBugAction.label"
-              menubarPath="help/helpStart"
-              style="push"
-              tooltip="%ReportBugAction.tooltip"/> 
-    	</actionSet>
+   <extension
+         point="org.eclipse.ui.commands">
+      <command
+            description="Report Bug or Enhancement for predefined Products / Projects"
```
Comment (ruspl-afed) [inline mylyn.tasks/org.eclipse.mylyn.tasks.bugs/plugin.xml:37]: Would be great to i18n this attribute

**Review by ruspl-afed — APPROVED** [review]
Thank you for reworking this PR @BeckerFrank  
Added two more comments to consider.

**Comment by merks** [comment]
Thanks for working on this and improving the implementation.

**Review by ruspl-afed — APPROVED** [review]
LGTM

**Pull request merged**